### PR TITLE
feat(graph): support creation of resource type allow/deny lists

### PR DESCRIPTION
**Pull request by gruebel**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

- added the possibility to define an allow/deny list of resource types

ex.

```yaml
---
metadata:
  name: "Ensure no Quantum Ledger Databases are provisoned"
  id: "CKV2_CUSTOM_1"
  category: "CONVENTION"
definition:
  cond_type: resource
  resource_types:
    - aws_qldb_ledger
  operator: not_exists
```

```hcl
resource "aws_qldb_ledger" "ledger" {
  name             = "example"
  permissions_mode = "STANDARD"
}

resource "aws_instance" "web" {
  ami           = data.aws_ami.ubuntu.id
  instance_type = "t3.micro"
}
```

output

```shell
Check: CKV2_CUSTOM_1: "Ensure no Quantum Ledger Databases are provisoned"
	PASSED for resource: aws_instance.web
	File: /main.tf:63-66
Check: CKV2_CUSTOM_1: "Ensure no Quantum Ledger Databases are provisoned"
	FAILED for resource: aws_qldb_ledger.ledger
	File: /main.tf:58-61

		58 | resource "aws_qldb_ledger" "ledger" {
		59 |   name             = "sample-ledger"
		60 |   permissions_mode = "STANDARD"
		61 | }
```

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/3.Custom Policies/YAML Custom Policies.md [hunk 1]
```diff
@@ -252,6 +253,22 @@ definition:
 
 *Note: The condition above uses AND logic. See [additional examples](https://www.checkov.io/3.Custom%20Policies/Examples.html) for complex logic in policy definitions.*
 
+## Resource Type Blocks
+
+A **Resource Type Block** in a policy's definition indicates that a resource will be compliant/non-complaint depending on the resource type, which is allowed/forbidden.
```
Comment (tsmithv11) [inline docs/3.Custom Policies/YAML Custom Policies.md:258]:
```suggestion
A **Resource Type Block** in a policy's definition indicates that a resource will be compliant/non-complaint depending on the resource type, which is allowed/forbidden. Use the `exist` operator to define an allowlist and the `not_exist` operator to define a blocklist.
```

File: docs/3.Custom Policies/YAML Custom Policies.md [hunk 2]
```diff
@@ -58,7 +58,8 @@ The top level object under `definition` must be a single object (not a list). It
 ## Types of Definition Blocks
 
 * **Attribute Blocks:** The policy describes resources with a certain configuration as defined by a configuration **attribute** and its value (per Terraform), or by the presence/absence of an attribute.
-* **Connection State Blocks** - The policy describes resources in a particular **Connection state**; either connected or not connected to another type of resource (for example, a security group).
+* **Connection State Blocks:**  The policy describes resources in a particular **Connection state**; either connected or not connected to another type of resource (for example, a security group).
+* **Resource Type Blocks:** The policy describes resources types, which are allowed or forbidden to use, so called allow/deny lists.
```
Comment (tsmithv11) [inline docs/3.Custom Policies/YAML Custom Policies.md:62]:
```suggestion
* **Resource Type Blocks:** The policy describes resource types that are either allowed or forbidden to use, commonly referred to as allow/deny lists.
```

**Review by tsmithv11 — APPROVED** [review]
🙌

File: tests/terraform/graph/checks_infra/resource_solvers/exists_solver/test_solver.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+from parameterized import parameterized_class
+
+from tests.terraform.graph.checks_infra.test_base import TestBaseSolver
+
+TEST_DIRNAME = Path(__file__).parent
+
+
+@parameterized_class([{"graph_framework": "NETWORKX"}, {"graph_framework": "IGRAPH"}])
+class ExistsSolver(TestBaseSolver):
+    def setUp(self):
```
Comment (bo156) [inline tests/terraform/graph/checks_infra/resource_solvers/exists_solver/test_solver.py:12]: maybe use `pytest` instead of `unittest` for new tests?

**Review by bo156 — APPROVED** [review]

File: tests/terraform/graph/checks_infra/resource_solvers/exists_solver/test_solver.py (hunk 3 above)
Comment (gruebel) [reply tests/terraform/graph/checks_infra/resource_solvers/exists_solver/test_solver.py:12]: the parent calss `TestBaseSolver` comes with a couple of convenient functions, which I didn't plan to duplicate, but I will double check what's the effort 🙂

**Comment by Saarett** [comment]
@gruebel @tsmithv11 @bo156 Are we planning on merging this PR? If not, I'll close it. Otherwise, let's prioritize and make it work

**Comment by tsmithv11** [comment]
@Saarett it was finished up here: https://github.com/bridgecrewio/checkov/pull/6451, so we can close this one.

**Comment by tsmithv11** [comment]
Thanks @gruebel for the help!

**Pull request closed without merging**